eventio/simtel/simtelfile.py

```diff
@@ -189,12 +189,14 @@ def _read_next_event(self):
 
         raise TypeError(f'Unexpected object found {self._file.peek()} in {self.path}')
 
-    def _check_skip(self, event):
+    def _check_keep(self, event):
+        '''check if we want to keep this event or throw it away'''
+
         if event['type'] == 'data':
-            return self.skip_non_triggered and not event.get('telescope_events')
+            return self.keep_non_triggered or event.get('telescope_events')
```

Comment:
This might now return the telescope events instead of a bool. It might not be important, but could be strange.

Reply:
oh!

Reply:
I've added `is_event_triggered = bool(event.get('telescope_events'))` ... since I must admin in the beginning I did not immediately understand why we get the telescope events there. :D

I think that fixes the issue you found and might even improve readability... what do you think?